packages/forms/src/elements/Checkbox.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import useFieldContext from '../utils/useFieldContext';
+import { CheckboxContext } from '../utils/useCheckboxContext';
+import { StyledCheckInput } from '../styled';
+
+/**
+ * Must be rendered within a `<Field>` element; accepts all
+ * `<input type="checkbox">` attributes and events.
+ */
+const Checkbox = React.forwardRef(({ indeterminate, children, ...props }, ref) => {
+  const { getInputProps } = useFieldContext();
+  const inputRef = inputElement => {
+    inputElement && (inputElement.indeterminate = indeterminate);
+  };
+  const combinedRef = inputElement => {
```

Comment:
Just leaving a note for us to revisit why the `useCombinedRefs` utility didn't work for this use-case.

Reply:
Yeah, meant to mention. In `useCombinedRefs` the `useEffect` messes with the timing of applying `indeterminate` to the underlying DOM node. The approach taken here can be seen in https://github.com/smooth-code/react-merge-refs/blob/master/src/index.js and might be a case for adding a `mergeRefs` utility.